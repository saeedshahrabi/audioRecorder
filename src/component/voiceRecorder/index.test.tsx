import { create } from 'react-test-renderer';
import { render, fireEvent, waitForElementToBeRemoved } from '@testing-library/react';
import VoiceRecorder from '.';


describe('record component', () => {
    const renderComponent = () => create(<VoiceRecorder  />);

    it('should render VoiceRecorder component', () => {
        const component = renderComponent().toJSON();
        expect(component).toMatchSnapshot()
    });
    it('should fire record button',() => {
        const { getByTestId } = render(<VoiceRecorder />);
        const record = getByTestId('record');
        expect(record).toBeInTheDocument()
        expect(record).toBeTruthy()
        fireEvent.click(record);
        waitForElementToBeRemoved(record)
    });
    
});
