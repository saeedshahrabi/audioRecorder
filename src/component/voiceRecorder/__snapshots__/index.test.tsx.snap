// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`record component should render VoiceRecorder component 1`] = `
<div
  className="recorder_container"
>
  <button
    data-testid="record"
    onClick={[Function]}
  >
    Record
  </button>
</div>
`;
